scipy/stats/_continuous_distns.py

```diff
@@ -832,8 +832,27 @@ def fit(self, data, *args, **kwds):
         return a, b, floc, fscale
 
     def _entropy(self, a, b):
-        return (sc.betaln(a, b) - (a - 1) * sc.psi(a) -
-                (b - 1) * sc.psi(b) + (a + b - 2) * sc.psi(a + b))
+        def regular(a, b):
+            return (sc.betaln(a, b) - (a - 1) * sc.psi(a) -
+                    (b - 1) * sc.psi(b) + (a + b - 2) * sc.psi(a + b))
+
+        def asymptotic_ab_large(a, b):
+            sum_ab = a + b
+            log_term = 0.5 * (
+                np.log(2 * np.pi) + np.log(a) + np.log(b) - 3 * np.log(sum_ab) + 1
+            )
+            t1 = (
+                (110 * sum_ab**3 + 20 * sum_ab**2 + sum_ab - 2)
+                * sum_ab**-4.0 / 120
+            )
+            t2 = (1 - 50 * a**3 - 10 * a**2 - a) * a**-4.0 / 120
+            t3 = (1 - 50 * b**3 - 10 * b**2 - b) * b**-4.0 / 120
+            return log_term + t1 + t2 + t3
```

Comment:
Thanks, but the point was that *all* positive powers and multiplications can cause premature overflow. When it's just as easy to avoid them, we might as well. Please double check that this is equivalent:
```suggestion
            ab = a + b
            log_term = 0.5 * (np.log(2*np.pi) + np.log(a) + np.log(b)
                              - 3*np.log(ab) + 1)
            t1 = 110/ab + 20*ab**-2. + ab**-3. - 2*ab**-4.
            t2 = -50/a - 10*a**-2. - a**-3. + a**-4.
            t3 = -50/b - 10*b**-2. - b**-3. + b**-4.
            return log_term + (t1 + t2 + t3)/120
```

Reply:
Oh I see. Thank you for the clarification.